### DISCUSSION TITLE
Rql parameters

### DISCUSSION THREAD

**Discussion by poostwoud** [ravendb/ravendb#14624] [General]
Hi, 

Is there a way to support parameters in rql and allow user input this way?

Cheers!

Peter

**Comment by ayende** [comment]
RQL supports parameters, certainly, you can use `$foo` as a parameter indicator.

Can you show us your current code? For Linq, it happens automatically.
For other scenarios, you call `AddParameter()`

**Reply by poostwoud** [reply]
Ah, missed that one! I'm using RawQuery.

```csharp
            var result = _session
                .Advanced
                .RawQuery<TDocument>(query)
                .AddParameter("$foo", foo)
                .Skip((currentPage - 1) * pageSize)
                .Take(pageSize)
                .ToList();
```

Case closed. Cheers!
